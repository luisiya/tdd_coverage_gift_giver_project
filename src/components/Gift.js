import React, { Component } from "react";
import { Form, FormGroup, FormControl, Button } from "react-bootstrap";

class Gift extends Component{
    constructor(props){
        super(props);
        this.state = {person: "", present: ""};
    }
    render(){
        return(
            <div className="gift">
               <Form>
                   <FormGroup>
                       <Form.Label>Person</Form.Label>
                       <FormControl
                           className="input_person"
                           onChange = {event => this.setState({person: event.target.value})}
                       />
                   </FormGroup>
                   <FormGroup>
                       <Form.Label>Present</Form.Label>
                       <FormControl
                           className="input_present"
                           onChange = {event => this.setState({present: event.target.value})}
                       />
                   </FormGroup>
               </Form>
                <Button
                 className="btn_remove"
                 onClick={() => this.props.removeGift(this.props.gift.id)}
                >
                    Remove Gift
                </Button>
            </div>
        )
    }
}
export default Gift;