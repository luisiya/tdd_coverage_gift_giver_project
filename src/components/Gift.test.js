import React from "react";
import {shallow} from "enzyme";
import Gift from "./Gift";

describe("Gift", () => {
    const mockRemove = jest.fn();
    const id = 1;
    const props = {gift: {id}, removeGift: mockRemove };
    const gift = shallow(<Gift {...props}/>);
    it("Rendered Gift correctly", () => {
        expect(gift).toMatchSnapshot();
    });

    it("Empty initial state for Gift", () => {
        expect(gift.state()).toEqual({person: "", present: ""});
    });
    describe("When typing some text in the Person field", () => {
        const person = "Uncle";
        beforeEach(() => {
            gift.find(".input_person").simulate("change", {target: {value: person}});
        });

        it("Update the Person in state", () => {
            expect(gift.state().person).toEqual(person);
        });

    });

    describe("When typing some text in the Present field", () => {
        const present = "BMW";
        beforeEach(() => {
            gift.find(".input_present").simulate("change", {target: {value: present}});
        });

        it("Update the Present in state", () => {
            expect(gift.state().present).toEqual(present);
        });
    });

    describe("When clicking the remove gift button", () => {
        beforeEach(() => {
            gift.find(".btn_remove").simulate("click");
        });
        it("call the remove function callback", () => {
            expect(mockRemove).toHaveBeenCalledWith(id);
        })
    })


});