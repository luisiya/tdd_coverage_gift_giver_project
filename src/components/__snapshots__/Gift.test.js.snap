// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift Rendered Gift correctly 1`] = `ShallowWrapper {}`;

exports[`Gift Rendered correctly 1`] = `ShallowWrapper {}`;
